nixos/modules/services/networking/fastnetmon-advanced.nix

```diff
@@ -0,0 +1,248 @@
+
+        fcli create_configuration
+        fcli delete hostgroup global
+        fcli import_configuration ${config_tar}
+        systemctl --no-block try-restart fastnetmon
+      '';
+      serviceConfig.Type = "oneshot";
+    };
+
+    systemd.services.fastnetmon-advanced = {
+      wantedBy = [ "multi-user.target" ];
+      after = [ "postgresql.service" "ferretdb.service" "fastnetmon-setup.service" ];
+      path = with pkgs; [ iproute2 ];
+      unitConfig = {
+        # Disable logic which shuts service when we do too many restarts
+        # We do restarts from sudo fcli commit and it's expected that we may have many restarts
+        # Details: https://github.com/systemd/systemd/issues/2416
+        StartLimitInterval = 0;
+      };
+      environment = {
+        LD_PRELOAD = "${pkgs.libredirect}/lib/libredirect.so";
+        NIX_REDIRECTS = "/etc/fastnetmon/license.lic=/var/lib/fastnetmon/license.lic:/etc/fastnetmon/gobgpd.conf=/run/fastnetmon/gobgpd.conf";
+      };
+      serviceConfig = {
+        ExecStart = "${pkgs.fastnetmon-advanced}/bin/fastnetmon --daemonize";
+
+        Type = "forking";
+        PIDFile="/run/fastnetmon/fastnetmon.pid";
+        LimitNOFILE = 65535;
+        # Restart service when it fails due to any reasons, we need to keep processing traffic no matter what happened
+        Restart= "on-failure";
+        RestartSec= "5s";
+
+        DynamicUser = true;
+        LogsDirectory = "fastnetmon";
+        CacheDirectory = "fastnetmon";
+        RuntimeDirectory = "fastnetmon"; # for pidfile
+        StateDirectory = "fastnetmon"; # for license file and gobgpd config
+      };
+    };
+
+    security.polkit.enable = true;
+    security.polkit.extraConfig = ''
+      polkit.addRule(function(action, subject) {
+        if (action.id == "org.freedesktop.systemd1.manage-units" &&
+          subject.isInGroup("fastnetmon")) {
+          if (action.lookup("unit") == "gobgp.service") {
+            var verb = action.lookup("verb");
+            if (verb == "start" || verb == "stop" || verb == "restart") {
+              return polkit.Result.YES;
+            }
+          }
+        }
+      });
+    '';
+
+    systemd.services.gobgp = {
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+      description = "GoBGP Routing Daemon";
+      unitConfig = {
+        ConditionPathExists = "/run/fastnetmon/gobgpd.conf";
+      };
+      serviceConfig = {
+        Type = "notify";
+        ExecStartPre = "${pkgs.fastnetmon-advanced}/bin/fnm-gobgpd -f /run/fastnetmon/gobgpd.conf -d";
+        SupplementaryGroups = [ "fastnetmon" ];
+        ExecStart = "${pkgs.fastnetmon-advanced}/bin/fnm-gobgpd -f /run/fastnetmon/gobgpd.conf --sdnotify";
+        ExecReload = "${pkgs.fastnetmon-advanced}/bin/fnm-gobgpd -r";
+        DynamicUser = true;
+        AmbientCapabilities = "cap_net_bind_service";
+      };
+    };
+  })
```

Comment:
There is a service for this in nixos

Reply:
Unsure if it is applicable. fastnetmon and gobgp are tightly integrated. We use the (older) gobgp binaries that fastnetmon ships, and fastnetmon generates the gobgp config at runtime while the gobgp module does not provide a way to specify a config file path (it is always a declarative config living in the nix store).

Reply:
Ah yes, I understand.

Reply:
fastnetmon-advanced has upgraded its bundled gobgp version to 3.12 (so finally 3.x!). We could check if the latest 3.x works as well.

Reply:
@JulienMalka do you still have concerns about this or are you fine if we leave this as is?

Reply:
So, we have integration tests for this, therefore we could try to unvendor this. Would be good for BGP security related updates.

Reply:
I'm not saying no if you want to invest the extra effort :D

Reply:
Probably throwing this into the endless backlog, I am fine with both options, @yu-re-ka might just want to pack this thing up :D

Reply:
I'm against this because the interaction between fastnetmon advanced and gobgp is critical and I don't trust our tests to cover every use case (like flowspec announcements), and I don't know how stable the gobgp api is.